wagtail_live/static/wagtail_live/js/intervalpolling.js

```diff
@@ -0,0 +1,201 @@
+ * current live posts appearing in the DOM in order to know which 
+ * live posts needs to be deleted.
+ */
+ class LivePostsTracker {
+    /**
+     * Creates a new LivePostsTracker.
+     * Initializes current live posts to an empty set.
+     * @constructor
+     */
+    constructor() {
+        this.currentLivePosts = new Set();
+    }
+
+    /**
+     * Sets current live posts from a list of live posts.
+     * @param {Array} livePosts - New live posts that replaces the previous ones.
+     */
+    setLivePosts(livePosts) {
+        this.currentLivePosts = new Set(livePosts);
+    }
+
+    /**
+     * Retrieves the live posts to delete.
+     * @param {Array} newLivePosts - Represents the current live posts on this page 
+     *  (some of which may not appear in the DOM yet).
+     * @returns {Array} containing the IDs of the live posts to delete.
+     */
+    getLivePostsToDelete(newLivePosts) {
+        let postsToDelete = [];
+
+        /** A live post has been deleted if it's in currentLivePosts and not in newLivePosts */
+        this.currentLivePosts.forEach(post => {
+            if (!newLivePosts.includes(post)) {postsToDelete.push(post)};
+        });
+
+        /** Set current live posts to new live posts. */
+        this.setLivePosts(newLivePosts);
+
+        return postsToDelete;
+    }
+}
+
+const basePollingURL = `/wagtail_live/get-updates/${channelID}/`;
+const SHAKING_INTERVAL = 1000;
+let POLLING_INTERVAL;
+let lastUpdateReceivedAt;
+let livePostsTracker = new LivePostsTracker();
+
+
+/**
+ * Initiates communication with server side.
+ * If response status is 200, this function initializes the current live posts, 
+ * the polling interval and the timestamp of the last update received then it waits
+ * for the duration of the polling interval and calls getUpdates.
+ * If response status isn't 200, it waits for the duration of the shaking interval and tries again.
+ */
+async function shake() {
+    let response = await fetch(basePollingURL, {
+        headers: {'X-CSRFToken': csrftoken},
+        method: 'POST',
+    });
+    if (response.status != 200) {
+        setTimeout(async () => await shake(), SHAKING_INTERVAL);
+    } else {
+        let [livePosts, lastUpdateTs, pollingInterval] = await response.json();
+        livePostsTracker.setLivePosts(livePosts);
+        [lastUpdateReceivedAt, POLLING_INTERVAL] = [lastUpdateTs, pollingInterval];
+        setTimeout(async () => await getUpdates(), POLLING_INTERVAL);
+    }
+}
+
+/**
+ * Main function which handles getting updates from the server side.
+ * @todo Needs better error handling.
+ */
+async function getUpdates() {
+    /** Retrieve timestamp of the last update of this page. */
+    let response = await fetchLastUpdateAt();
+    if (response.status != 200) {
+        setTimeout(async () => await getUpdates(), POLLING_INTERVAL);
+    }
+    else {
+        /** Check if new updates are available. */
+        if (newUpdate(response)) {
```

Comment:
Thanks a lot for this suggestion. 

> You could even consider collapsing them into one condition, since they all seem to do the same (stop and wait for updates).

I don't know how to go for this. They all do the same but at different moments.